### ISSUE TITLE
Package with no `license` throws an Error when invoked with `--onlyAllow`

### ISSUE THREAD

**Issue by jonny-improbable** [davglass/license-checker#174]
## Version
I am using license-checker@22.2.0

## Usage
I am invoking license-checker with the following arguments:

```
 license-checker --onlyAllow="MIT;ISC"
```

One of the dependencies being tested does not contain a "license" field in its `package.json` file (package in question is [@segment/loosely-validate-event](https://github.com/segmentio/loosely-validate-event)

## Actual
The following uncaught exception is raised:

```
.../license-checker/lib/index.js:339
                    if (data[item].licenses.indexOf(k) === -1 && !good) {
                                            ^

TypeError: Cannot read property 'indexOf' of null
```

## Expected
license-checker should exit with status 1 and the error message: 

> Found license NOT defined by the --onlyAllow flag: "UNLICENSED". Exiting

**Comment by davglass** [comment]
Hmm, I can't seem to replicate this:

```
license-checker --onlyAllow="MIT;ISC"
├─ @segment/loosely-validate-event@2.0.0
│  ├─ licenses: MIT*
│  ├─ path: /Users/davglass/tmp/node_modules/@segment/loosely-validate-event
│  └─ licenseFile: /Users/davglass/tmp/node_modules/@segment/loosely-validate-event/LICENSE
├─ component-type@1.2.1
│  ├─ licenses: MIT
│  ├─ repository: https://github.com/component/type
│  ├─ path: /Users/davglass/tmp/node_modules/component-type
│  └─ licenseFile: /Users/davglass/tmp/node_modules/component-type/Readme.md
└─ join-component@1.1.0
   ├─ licenses: MIT
   ├─ path: /Users/davglass/tmp/node_modules/join-component
   └─ licenseFile: /Users/davglass/tmp/node_modules/join-component/Readme.md
```

**Comment by jonny-improbable** [comment]
Ah sorry, should have been clearer - issue is reproducible with @segment/loosely-validate-event@1.1.2 (note latest package version is 2.0.0 where a LICENSE file has been added).

```
$ license-checker --onlyAllow="MIT;ISC"
/.../license-checker/lib/index.js:339
                    if (!data[item].private && data[item].licenses.indexOf(k) === -1 && !good) {
                                                                   ^

TypeError: Cannot read property 'indexOf' of null
    at /.../license-checker/lib/index.js:339:68
```

**Comment by davglass** [comment]
Pushed in `license-checker@21.0.0`
